servlet.py

```diff
@@ -501,17 +501,27 @@ def translateAndRespond(self, pair, pipeline, toTranslate, markUnknown, nosplit=
 
     @gen.coroutine
     def get(self):
-        pair = self.getPairOrError(self.get_argument('langpair'),
-                                   len(self.get_argument('q')))
-        if pair is not None:
-            pipeline = self.getPipeline(pair)
-            deformat, reformat = self.getFormat()
-            yield self.translateAndRespond(pair,
-                                           pipeline,
-                                           self.get_argument('q'),
-                                           self.get_argument('markUnknown', default='yes'),
-                                           nosplit=False,
-                                           deformat=deformat, reformat=reformat)
+        self.response = []
+        query_list = self.get_arguments('q')
+        langpair = self.get_argument('langpair')
+        for query in query_list:
+            pair = self.getPairOrError(langpair, len(query))
+            if pair is not None:
+                pipeline = self.getPipeline(pair)
+                deformat, reformat = self.getFormat()
+                yield self.translateAndRespond(pair,
+                                               pipeline,
+                                               query,
+                                               self.get_argument('markUnknown', default='yes'),
+                                               nosplit=False,
+                                               deformat=deformat, reformat=reformat)
+        if len(self.response) == 1:
```

Comment:
This code never executes. `yield` is like `return`. I suggest you read up on how `yield` works and try running this code locally and `print`ing to see what order the statements are executed in.